be/src/olap/task/build_inverted_index.cpp

```diff
@@ -0,0 +1,461 @@
+}
+
+Status BuildInvertedIndex::handle_single_rowset(RowsetMetaSharedPtr output_rowset_meta,
+                                std::vector<segment_v2::SegmentSharedPtr>& segments) {
+    if (_is_drop_op) {
+        // delete invertd index file by gc thread when gc input rowset
+        return Status::OK();
+    } else {
+        // create inverted index writer
+        std::string segment_dir = _tablet->tablet_path();
+        auto fs = output_rowset_meta->fs();
+        auto output_rowset_schema = output_rowset_meta->tablet_schema();
+        for (auto& seg_ptr : segments) {
+            std::string segment_filename =
+                    fmt::format("{}_{}.dat", output_rowset_meta->rowset_id().to_string(), seg_ptr->id());
+            std::vector<ColumnId> return_columns;
+            std::vector<std::pair<int64_t, int64_t>> inverted_index_writer_signs;
+            _olap_data_convertor->reserve(_alter_inverted_indexes.size());
+            // create inverted index writer
+            for (auto i = 0; i < _alter_inverted_indexes.size(); ++i) {
+                auto inverted_index = _alter_inverted_indexes[i];
+                DCHECK_EQ(inverted_index.columns.size(), 1);
+                auto index_id = inverted_index.index_id;
+                auto column_name = inverted_index.columns[0];
+                auto col_uid = _alter_column_uids[i];
+                DCHECK(output_rowset_schema->has_inverted_index(col_uid));
+                auto column = output_rowset_schema->column_by_uid(col_uid);
+                _olap_data_convertor->add_column_data_convertor(column);
+                auto column_idx = output_rowset_schema->field_index(col_uid);
+                if (column_idx < 0) {
+                    LOG(WARNING) << "referenced column was missing. "
+                                 << "[column=" << column_name << " referenced_column=" << column_idx
+                                 << "]";
+                    return Status::Error<ErrorCode::CE_CMD_PARAMS_ERROR>();
+                }
+                return_columns.emplace_back(column_idx);
+
+                std::unique_ptr<Field> field(FieldFactory::create(column));
+                auto index_meta = output_rowset_schema->get_inverted_index(col_uid);
+                std::unique_ptr<segment_v2::InvertedIndexColumnWriter> inverted_index_builder;
+                try {
+                    RETURN_IF_ERROR(segment_v2::InvertedIndexColumnWriter::create(
+                            field.get(), &inverted_index_builder, segment_filename, segment_dir,
+                           index_meta, fs));
+                } catch (const std::exception& e) {
+                    LOG(WARNING) << "CLuceneError occured: " << e.what();
+                    return Status::Error<ErrorCode::IO_ERROR>();
+                }
+
+                if (inverted_index_builder) {
+                    auto writer_sign = std::make_pair(seg_ptr->id(), index_id);
+                    _inverted_index_builders.insert(
```

Comment:
We can process index one by one for each segment, to avoid _inverted_index_builders and read many columns in next_batch to save memory.